rfd/0109-cloud-agent-upgrades.md

```diff
@@ -327,6 +327,33 @@ to be changed to follow the same pattern of registering a release channel and in
 by ec2 discovery.
 
 
+### ECS Fargate Agents
+When Teleport Discover flow is used to connect an RDS, a Teleport agent is automatically provisioned using ECS Fargate. The automatic update
+functionality for regular agents depends on systemd or Kubernetes. Fargate agents run in containers, so that means the existing automatic
+update functionality is not compatible with these agents. We need a separate method of keeping these Fargate agents updated.
+
+To keep Farage agents updated, Teleport will run a separate task that checks every 30 minutes for an available update and uses the ECS
+Fargate API to update Teleport agents. This task does not need to be run on all Teleport clusters. It will only be needed if the
+`TELEPORT_AUTOMATIC_UPGRADES` environment variable is enabled. The window when Fargate agents are updated should also coincide with the
+tenant upgrade window.
+
+The logic flow will look like the same as the core update model:
+
+- If there is a critical update, attempt update
+- If an agent is unhealthy, attempt update
```

Comment:
How do we know the agent is unhealthy?

Reply:
Hmm, do you think we'd be able to use the service/task status to help determine if an agent is healthy/unhealthy?

Reply:
We do not, we must define the HealthCheck for the TaskDefinition.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-healthcheck.html

Then, we must have a command we can run to detect whether Teleport is correctly running (there's no other binary besides `teleport` within the container - we are using the distroless container).